fideslib/core/config.py

```diff
@@ -108,15 +108,27 @@ def validate_encryption_key_length(
             )
         return v
 
-    CORS_ORIGINS: List[AnyHttpUrl] = []
+    CORS_ORIGINS: List[str] = []
 
     @validator("CORS_ORIGINS", pre=True)
     @classmethod
     def assemble_cors_origins(cls, v: Union[str, List[str]]) -> Union[List[str], str]:
         """Return a list of valid origins for CORS requests"""
+
+        def validate(values: List[str]) -> None:
+            for value in values:
+                if value != "*":
+                    if not validators.url(value):
+                        raise ValueError(f"{value} is not a valid url")
```

Comment:
Because the field type for `CORS_ORIGINS` is now `List[str]` to be able to accept the * values we lose access to Pydantic's `AnyHttpUrl` validator. This is a custom validator to take its place.

Reply:
You are correct, `htttps://*.example.com/` would not be supported here. `allow_origini_regex` comes from starlette, https://github.com/encode/starlette/blob/ce0709d340fd313ae96a11abfc50972041394ebb/starlette/middleware/cors.py#L29. I we add that as middleware to FastAPI I think we could do away with the validation all together and the middlewear would handle it?

Reply:
We opted not to pursue regex at first as it's a higher burden for the person deploying Fidesops to understand, so let's stand by that choice for now.